packages/teleterm/src/ui/services/workspacesService/documentsService/documentsService.ts

```diff
@@ -157,11 +157,12 @@ export class DocumentsService {
       return;
     }
 
-    const nextUri = this.getNextUri(uri);
-    const docs = this.getState().documents.filter(d => d.uri !== uri);
     this.setState(draft => {
-      draft.documents = docs;
-      draft.location = nextUri;
+      if (draft.location === uri) {
+        draft.location = this.getNextUri(uri);
+      }
+
+      draft.documents = this.getState().documents.filter(d => d.uri !== uri);
```

Comment:
What problem does this solve? I have trouble imagining what was wrong with the previous code.

Reply:
In the previous implementation `location` was changed when any of documents was closed and that resulted in changing the active tab when you closed inactive tab. We want to change the `location` only when the currently active tab is closed.